xml/obs_build_config.xml

```diff
@@ -711,64 +790,131 @@ Keep: debbuild</screen>
   </variablelist>
  </sect1>
 
- <sect1 xml:id="sec.prjconfig.macros">
-  <title>Macro Section</title>
-  <remark>toms 2017-08-18: TODO: Extend this section from https://en.opensuse.org/openSUSE:Build_Service_prjconf#Macros</remark>
+ <sect1 xml:id="sec.prjconfig.ccache">
+  <title>Building with ccache or sccache</title>
   <para>
-   Macros are defined at the end of the project configuration.
-   The macro section is only used on RPM builds.
-   <!--
-    The content gets exported to be used by <command>rpmbuild</command>
-   command during the build.
-   It can be used to export a define using "%key value" lines.
-   -->
+    The usage of ccache or sccache can be enabled for each package by
+    seting the useccache:PACKAGE build flag.
   </para>
   <para>
-   The project configuration knows two possible definitions:
+    The ccache package will automatically be installed and configured.
+    The directory /.ccache/ will be configured as cache
+    directory. To configure ccache, the file /.ccache/ccache.conf can be
+    modified as part of the build process by the $BUILD_USER environment
+    variable.
+  </para>
+  <para>
+    In some cases, there is no archive for the current package, such as when
+    the package was newly branched or when binaries were deleted. In these
+    cases, the system will check whether there is a package of the same name
+    built for the same architecture within one of the repositories
+    configured in the project's meta configuration. If so, the archive of
+    that package will be used. The repositories will be searched in the
+    order they are configured in the meta configuration, starting from the
+    top.
+  </para>
+  <para>
+    An alternative way to enable caching based on build dependencies is to
+    add "--enable-cache" as dependency, for example via a Substitute rule:
+    <screen><command>
+        Substitute: gcc-c++ gcc-c++ --enable-ccache
+    </command></screen>
+    This will always enable ccache when a direct build depdency to gcc-c++
+    is required.
+  </para>
+  <para>
+    It is also possible to set the type, eg:
+    <screen><command>
+        Substitute: cargo cargo --enable-ccache=sccache
+    </command></screen>
+  </para>
```

Comment:
@tomschr Here's where the error got introduced. Merged on March 22.

Reply:
Just for the record and for the others: the default validation in our doc CI changed from DocBook to GeekoDoc. The latter is a subset of the former. It's basically a more simplified version of DocBook. The elements are the still the same, the set of elements are reduced and some content models are more restrictive.

In this particular case, GeekoDoc doesn't allow other block elements like `<screen>` inside a paragraph. You have two options to solve that:

1. Going back and validating with DocBook.
   In that case, add in your DC files the following line:

       DOCBOOK5_RNG_URI="http://docbook.org/xml/5.2/rng/docbookxi.rng"

2. Migrate your docs to validate with GeekoDoc.
   In most cases, validation errors with GeekoDoc falls into one of the following categories:

   * Wrong `xml:id` values
     Dots aren't allowed, only digits, lowercase and uppercase A-Z letters, and dashes.

   * Unknown elements
     Some elements like `<footnote>` or `<footnoteref>` were disabled. In this case, you can replace `<footnoteref>` with `<xref/>`. For footnotes, you need to find a substitution and need to restructure the content.

   * Block elements inside paragraphs
     DocBook distinguishes between division, block, and inline elements. Division elements contain block elements, and block elements inline elements. Sometimes you can nest block elements inside other block elements. GeekoDoc made that simpler and forbid this. You need to clearly separate them.

Which of the approach is the better solution depends on what features you need. GeekoDoc is a bit simpler, but could confuse people if they are used to DocBook or only know that.